lib/parallel_tests/test/runner.rb

```diff
@@ -209,23 +209,20 @@ def sort_by_filesize(tests)
         end
 
         def find_tests(tests, options = {})
-          files_list = (tests || []).map do |file_or_folder|
+          suffix_pattern = options[:suffix] || test_suffix
+          include_pattern = options[:pattern] || //
+          exclude_pattern = options[:exclude_pattern]
+
+          (tests || []).flat_map do |file_or_folder|
             if File.directory?(file_or_folder)
               files = files_in_folder(file_or_folder, options)
-              files.grep(options[:suffix] || test_suffix)
-                .grep(options[:pattern] || //)
+              files = files.grep(suffix_pattern).grep(include_pattern)
+              files -= files.grep(exclude_pattern) if exclude_pattern
```

Comment:
Moving the filtering of the exclusion into this block means folders are not filtered out.  If the goal is to mirror the functionality of rspec `--pattern` and `--exclude-pattern, wouldn't the folder need to have the regexes applied as well?

Reply:
it means files that are explicitly added are not affected by filters
folders are still effected since they are covered to files and then matched against